src/session_impl.cpp

```diff
@@ -461,15 +461,15 @@ void apply_deprecated_dht_settings(settings_pack& sett, bdecode_node const& s)
 int ssl_server_name_callback(SSL* s, int*, void* arg)
 {
 	char const* name = SSL_get_servername(s, TLSEXT_NAMETYPE_host_name);
-	session_impl* si = reinterpret_cast<session_impl*>(arg);
+	auto si = reinterpret_cast<session_impl*>(arg);
 	return ssl_server_name_callback_impl(s, name ? std::string(name) : "", si)
 			? SSL_TLSEXT_ERR_OK
 			: SSL_TLSEXT_ERR_ALERT_FATAL;
 }
 #elif defined TORRENT_USE_GNUTLS
 bool ssl_server_name_callback(ssl::stream_handle_type stream_handle, std::string const& name, void* arg)
 {
-	session_impl* si = reinterpret_cast<session_impl*>(arg);
+	auto si = reinterpret_cast<session_impl*>(arg);
```

Comment:
I really prefer to preserve the "pointerness" in auto expressions. i.e. `auto* si = ` (applies to this change and the one above)

Reply:
There is a `clang-tidy` check for this, btw: [readability-qualified-auto](https://clang.llvm.org/extra/clang-tidy/checks/readability-qualified-auto.html)

Reply:
Fixed.